src/linear_algebra/contraction.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2020 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import linear_algebra.tensor_product
+import linear_algebra.dual
+
+/-!
+# Contractions
+
+Given modules $M, N$ over a commutative ring $R$, this file defines the natural linear maps:
+$M^* \otimes M \to R$, $M \otimes M^* \to R$, and $M^* \otimes N → Hom(M, N)$, as well as proving
+some basic properties of these maps.
+
+## Tags
+
+contraction, dual module, tensor product
+-/
+
+universes u v
+
+set_option class.instance_max_depth 42
+
+namespace contraction
+open tensor_product
+open_locale tensor_product
+
+variables (R : Type u) (M N : Type v)
+variables [comm_ring R] [add_comm_group M] [add_comm_group N] [module R M] [module R N]
+
+/--
```

Comment:
I don't really care, but I think you can just put `/-- docstring -/` on one line, instead of three. You don't have to change it. Just a remark.

Reply:
Thanks, I'll change it. I had thought I was following some preferred style, and actually prefer compressing to fewer lines.